clients/rust/src/utils.rs

```diff
@@ -11,12 +16,29 @@ pub fn clean(value: String) -> String {
 
 /// Checks that the `master_edition` is Programmable NFT master edition.
 pub fn assert_edition_is_programmable(edition_data: &[u8]) -> Result<(), MplTokenMetadataError> {
-    if !edition_data.is_empty()
-        && edition_data[edition_data.len() - TOKEN_STANDARD_OFFSET]
-            == TokenStandard::ProgrammableNonFungible as u8
-    {
-        Ok(())
-    } else {
-        Err(MplTokenMetadataError::InvalidTokenStandard)
+    if !edition_data.is_empty() {
+        // the first byte is the account key
+        let key = Key::deserialize(&mut &edition_data[0..1])
+            .map_err(|_error| MplTokenMetadataError::InvalidEditionKey)?;
+
+        if matches!(key, Key::MasterEditionV1 | Key::MasterEditionV2) {
+            // the last byte is the token standard
+            let standard = TokenStandard::deserialize(
+                &mut &edition_data[edition_data.len() - TOKEN_STANDARD_OFFSET..],
```

Comment:
I probably still would recommend incorporating `if edition_data.len() >= TOKEN_STANDARD_OFFSET` but I think we plan for `TOKEN_STANDARD_OFFSET` to always just be 1.